awx/main/tasks.py

```diff
@@ -1049,6 +1049,29 @@ def build_env(self, instance, private_data_dir, isolated, private_data_files=Non
 
         env['AWX_PRIVATE_DATA_DIR'] = private_data_dir
 
+        # Note: This code will only work with Podman 3.1.0+
+        # which added CONTAINERS_REGISTRIES_CONF
+        ee_cred = self.instance.execution_environment.credential
+        if ee_cred:
+            verify_ssl = ee_cred.get_input('verify_ssl')
+            if not verify_ssl:
+                pdd_wrapper_path = os.path.split(private_data_dir)[0]
+                registries_conf_path = os.path.join(pdd_wrapper_path, 'registries.conf')
+                host = ee_cred.get_input('host')
+
+                with open(registries_conf_path, 'w') as registries_conf:
+                    os.chmod(registries_conf.name, stat.S_IRUSR | stat.S_IWUSR)
+
+                    lines = [
+                        '[[registry]]',
+                        'location = "{}"'.format(host),
+                        'insecure = true',
+                    ]
+
+                    registries_conf.write('\n'.join(lines))
```

Comment:
Random remark after poking around in the code: At first, I had a concern that it might be possible for this file to be unexpectedly used by future job runs after the `verify_ssl` flag gets set back to `true`.

After some testing and thinking, I realized that isn't a problem because the job runs are isolated and these directories aren't reused

![Screenshot from 2021-04-19 19-40-53](https://user-images.githubusercontent.com/9753817/115406570-4bf26600-a1bd-11eb-845f-bfd77fb07a60.png)


... Then, after some more thinking, I realized that _even if these directories were reused_, it _still_ wouldn't be a problem because `env['CONTAINERS_REGISTRIES_CONF']` is only set when `verify_ssl` is false.